core/runtime/src/main/java/io/quarkus/runtime/BuildAnalyticsConfig.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * Build time analytics configuration.
+ * This is a dummy config class to hide the warnings on the comment line.
+ * All properties in here are actually used in the build tools.
+ */
+@ConfigRoot(name = "analytics", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class BuildAnalyticsConfig {
+
+    /**
+     * If Build time analytics are disabled.
+     */
+    @ConfigItem
+    public Optional<Boolean> disabled;
+
+    /**
+     * The Segment base URI.
+     */
+    @ConfigItem
+    public Optional<String> uriBase;
+
+    /**
+     * The Timeout to send the build time analytics to segment.
+     */
+    @ConfigItem(defaultValue = "3000")
+    public Optional<Integer> timeout;
```

Comment:
Let's NOT push this PR again for that but could you get rid of the `Optional` in a future version given we have a default value?

But let's merge this before iterating on that.

Reply:
sure

Reply:
On a follow up PR